pkg/controller/controller_instance.go

```diff
@@ -107,196 +234,119 @@ func (c *controller) reconcileInstance(instance *v1alpha1.Instance) error {
 		return c.pollInstance(serviceClass, servicePlan, brokerName, brokerClient, instance)
 	}
 
-	if instance.DeletionTimestamp == nil { // Add or update
-		glog.V(4).Infof("Adding/Updating Instance %v/%v", instance.Namespace, instance.Name)
-
-		var parameters map[string]interface{}
-		if instance.Spec.Parameters != nil {
-			parameters, err = unmarshalParameters(instance.Spec.Parameters.Raw)
-			if err != nil {
-				s := fmt.Sprintf("Failed to unmarshal Instance parameters\n%s\n %s", instance.Spec.Parameters, err)
-				glog.Warning(s)
-				c.updateInstanceCondition(
-					instance,
-					v1alpha1.InstanceConditionReady,
-					v1alpha1.ConditionFalse,
-					errorWithParameters,
-					"Error unmarshaling instance parameters. "+s,
-				)
-				c.recorder.Event(instance, api.EventTypeWarning, errorWithParameters, s)
-				return err
-			}
-		}
+	glog.V(4).Infof("Adding/Updating Instance %v/%v", instance.Namespace, instance.Name)
```

Comment:
Not feedback about your PR, just a signpost: this is where we will have to determine whether we're doing an add or an update.